### ISSUE TITLE
[CoE Starter Kit - BUG] Update to the latest solution failed

### ISSUE THREAD

**Issue by ValiPrk** [microsoft/coe-starter-kit#8619]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

When trying to install the newest Center of Excellence - Core Components solution (4.33) I receive an error: Solution "Center of Excellence - Core Components" failed to import: Solution Upgrade action failed after import as holding. InnerException is: Solution dependencies exist, cannot uninstall. DependencyCount : 1 RequiredComponentObject details: Type: Entity, ObjectName: DEPRECATED: COE Settings, Id: a4e1a204-2180-4051-81c1-f175b943496f, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: SiteMap, ObjectName: cr913_Test2107ModelDriven, Id: 178112d0-eb08-ed11-82e5-0022487f06b2, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published .

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.32.2

### What app or flow are you having the issue with?

CenterofExcellenceCoreComponents

### What method are you using to get inventory and telemetry?

None

**Comment by ValiPrk** [comment]
Hi, 

Thank you for your reply, I found the mentioned site map in the Default Solution of the CoE environment. I checked the dependencies and there were none (see attached screenshots).
Can you please tell me what would be the next steps?
Remove the site map from the solution? How to proceed with the faulty version 4.33 of the Core Components solution ? Should it be deleted?
![Dependencies1](https://github.com/microsoft/coe-starter-kit/assets/147488275/4e02b72a-2ae6-43f8-8bf6-49d546c99a68)
![Dependencies](https://github.com/microsoft/coe-starter-kit/assets/147488275/8fe21b9d-b292-4cc3-821e-45f12524b76a)
![Site map](https://github.com/microsoft/coe-starter-kit/assets/147488275/00c61538-b664-48b6-a75c-7ab7de525722)

**Comment by Jenefer-Monroe** [comment]
It looks like its caslled

**Comment by ValiPrk** [comment]
Hello, it was underneath the one that send the first screenshot, the "Delete blocked by" and "Used by" columns are empty, the only data is under the "Uses" column (see attached screenshot).
I did not create it, it's on the Default solution on the CoE environment, this is why I'm asking for the next steps, I would not want to delete something that will have impact.
Thank you for your collaboration
![Site map correct](https://github.com/microsoft/coe-starter-kit/assets/147488275/2d4b2ef0-c117-4fb6-ab27-f0ebf6ee6082)

**Comment by ValiPrk** [comment]
Also found the table, see attached screenshots with dependencies
![Dependencies3](https://github.com/microsoft/coe-starter-kit/assets/147488275/ac6ed72c-4072-4021-80ea-d0068560d7a1)
![Dependencies2](https://github.com/microsoft/coe-starter-kit/assets/147488275/1dc9d7c6-2be4-4cfc-91d5-63c128cf5257)

**Comment by Jenefer-Monroe** [comment]
The first screenshot you sent without the dependency is 1207
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/2e55cf6f-5020-4578-9fd4-bb88f3b2e2e3)

Whereas the dependency is in 2107
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/ddd7ce41-b648-49fe-b0c8-7526273a36c0)

So you'll need to find that app / sitemap and clean them up to not rely on this table.

**Comment by Jenefer-Monroe** [comment]
I'm afraid I can't comment on exactly how to remove all these dependencies as I am not looking into your tenant. 

I can't imagine that this is what is blocking you but you can try. 

That settings icon web resource is used elsewhere in the kit as you show so I dont think deleting it is the correct path forward. 
You can remove the table's dependency on it though
This is the item you are removing from the table
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/daad7d7d-ee74-4a21-9632-c7a5740ad3a8)

Just set it to none
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/6cea26d4-c5f2-4c6c-80a0-54c4543f1fbc)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/c802da2e-ca95-48db-b0d5-4ab159899fa5)

**Comment by Jenefer-Monroe** [comment]
What is needed is to open that site map and remove the reference to this table. Perhaps you've already done that? 
You'll know you're ready to try again when the dependencies show blank
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/4ca80190-f71f-4e64-9fa5-166d4ee37759)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/44bb8c80-5dfd-4306-a31e-32671d2013a2)

**Comment by Jenefer-Monroe** [comment]
Then correct, you will delete the _Upgrade and try again once you are there.

**Comment by ValiPrk** [comment]
Hi, 

I changed the table settings as suggested and indeed, if I'm checking the Settings icon [Customization.Type_WebResource] it doesn't show DEPRECATED COE Setting as a dependency, so far so good.
When I'm going to the "Test2107ModelDriven" sitemap, the "remove dependency" option is greyed out (see attached screenshot).
 
![COE settings](https://github.com/microsoft/coe-starter-kit/assets/147488275/99ec827f-f1ae-46f5-985f-8ab3c60425a3)
![remove greyed-out](https://github.com/microsoft/coe-starter-kit/assets/147488275/bdb537e4-4fe9-4458-a684-78c848807eb6)

**Comment by ValiPrk** [comment]
Hi,

I managed to get rid of the sitemap by removing it from the table (that was already empty). This I was able to install the newest solution. I checked the flows, the audit logging, everything is working and it seems to be in order.
Thank you very much for your help and patience! 
Wish you a great weekend!
Valentin
